src/Network/NWFramer.cs

```diff
@@ -180,21 +180,22 @@ static void TrampolineOutputHandler (IntPtr block, OS_nw_framer framer, OS_nw_pr
 		[DllImport (Constants.NetworkLibrary)]
 		unsafe static extern void nw_framer_set_input_handler (OS_nw_framer framer, void *input_handler);
 
-		delegate void nw_framer_set_input_handler_t (IntPtr block, OS_nw_framer framer);
+		delegate int nw_framer_set_input_handler_t (IntPtr block, OS_nw_framer framer);
 		static nw_framer_set_input_handler_t static_InputHandler = TrampolineInputHandler;
 
 		[MonoPInvokeCallback (typeof (nw_framer_set_input_handler_t))]
-		static void TrampolineInputHandler (IntPtr block, OS_nw_framer framer)
+		static int TrampolineInputHandler (IntPtr block, OS_nw_framer framer)
 		{
-			var del = BlockLiteral.GetTarget<Action<NWFramer>> (block);
+			var del = BlockLiteral.GetTarget<Func<NWFramer, int>> (block);
 			if (del != null) {
 				var nwFramer = new NWFramer (framer, owns: true);
-				del (nwFramer);
+				return del (nwFramer);
 			}
+			return 0;
 		}
 
 		[BindingImpl (BindingImplOptions.Optimizable)]
-		public Action<NWFramer> InputHandler {
+		public Func<NWFramer, int> InputHandler {
```

Comment:
`Func<NWFramer, int>` hard to guess that the `int` is without reading the docs
and the best way not to write docs is to have a custom delegate :)